packages/@expo/metro-config/src/ExpoMetroConfig.ts

```diff
@@ -338,6 +338,7 @@ export function getDefaultConfig(
       ),
       assetRegistryPath: '@react-native/assets-registry/registry',
       assetPlugins: getAssetPlugins(projectRoot),
+      hermesParser: true,
```

Comment:
@byCedric I found that if we don't enable this bunch of react-native internal files that use Flow will just error out.
 
<img width="1105" alt="Screenshot 2024-09-26 at 18 01 21" src="https://github.com/user-attachments/assets/119ef4df-6302-4f2e-acdd-d1431617a5c6">